examples/model_selection/plot_grid_search_digits.py

```diff
@@ -13,56 +13,177 @@
 
 More details on tools available for model selection can be found in the
 sections on :ref:`cross_validation` and :ref:`grid_search`.
-
 """
 
+# %%
+# The dataset
+# -----------
+#
+# We will work with the `digits` dataset. The goal is to classify handwritten
+# digits images.
 from sklearn import datasets
-from sklearn.model_selection import train_test_split
-from sklearn.model_selection import GridSearchCV
-from sklearn.metrics import classification_report
-from sklearn.svm import SVC
 
-# Loading the Digits dataset
-X, y = datasets.load_digits(return_X_y=True)
+digits = datasets.load_digits()
+
+# %%
+# To apply a classifier on this data, we need to flatten the images.
+# Each image of 8 by 8 pixels needs to be transformed to a vector of 64 pixels.
+# Thus, we will get a final data array of shape (n_images, n_pixels)
+n_samples = len(digits.images)
+X = digits.images.reshape((n_samples, -1))
+y = digits.target == 7
+print(
+    f"The number of images is {X.shape[0]} and each image contains {X.shape[1]} pixels"
+)
+
+# %%
+# As presented in the introduction, the data will be split into a development
+# and evaluation set of equal size.
+from sklearn.model_selection import train_test_split
 
-# Split the dataset in two equal parts
 X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.5, random_state=0)
 
-# Set the parameters by cross-validation
+# %%
+# Define our grid-search framework
+# --------------------------------
```

Comment:
I think, since `refit_strategy` is used, it makes sense to use the same term here.
```suggestion
# Define our grid-search strategy
# -------------------------------
```